### PR TITLE
[QE] Edit Provider infra and test cases for vmware and rhev providers

### PR DESCRIPTION
**Pull request by kpunwatk**
1. Added edit plan method in RHV provider 
2. Modified vmware edit provider method
3. Modified both rhv and vmware configs according to the edit provider data
4. Added test cases 	

modified:   pkg/qe-tests/cypress/integration/models/providerVmware.ts
	modified:   pkg/qe-tests/cypress/integration/tests/Rhv/config_separate_mapping_rhv.ts
	new file:   pkg/qe-tests/cypress/integration/tests/Rhv/editProvider.ts
	modified:   pkg/qe-tests/cypress/integration/tests/vmware/config_separate_mapping.ts
	new file:   pkg/qe-tests/cypress/integration/tests/vmware/editProvider.ts
	modified:   pkg/qe-tests/cypress/integration/types/types.ts

### CODE REVIEW

File: pkg/qe-tests/cypress/integration/models/providerVmware.ts [hunk 1]
```diff
@@ -153,16 +153,18 @@ export class ProviderVmware extends Provider {
...
   }
 
-  edit(providerData: VmwareProviderData): void {
-    const { name, hostname, username, password } = providerData;
+  edit(providerData: VmwareProviderData, editProviderData: VmwareProviderData): void {
+    const { name } = providerData;
+    const { hostname, username, password } = editProviderData;
     ProviderVmware.openList();
     applyAction(name, editButton);
     this.fillHostname(hostname);
```
Comment (sshveta) [inline pkg/qe-tests/cypress/integration/models/providerVmware.ts:161]: Will the user always send updated values for all these fields ? Hostname, username ?
What if I just want to update hostname and not rest of the fields ?

File: pkg/qe-tests/cypress/integration/tests/Rhv/editProvider.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { editProviderData, testData } from './config_separate_mapping_rhv';
+import { login } from '../../../utils/utils';
+import { providerRhv } from '../../models/providerRhv';
+import { RhvProviderData } from '../../types/types';
+
+describe('Creating provider and deleting', () => {
```
Comment (sshveta) [inline pkg/qe-tests/cypress/integration/tests/Rhv/editProvider.ts:6]: CRUD (Create, Read, Update and Delete) operations is a better description for this test suite .

describe('Provider CRUD Operations', () => {

File: pkg/qe-tests/cypress/integration/tests/vmware/editProvider.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { editProviderData, testData } from './config_separate_mapping';
+import { login } from '../../../utils/utils';
+import { ProviderVmware } from '../../models/providerVmware';
+import { VmwareProviderData } from '../../types/types';
+
+describe('Creating provider and deleting', () => {
```
Comment (sshveta) [inline pkg/qe-tests/cypress/integration/tests/vmware/editProvider.ts:6]: Same as above , 

     describe(' Provider CRUD Operations ',() => {

**Review by sshveta — DISMISSED** [review]

File: pkg/qe-tests/cypress/integration/models/providerVmware.ts (hunk 1 above)
Comment (kpunwatk) [reply pkg/qe-tests/cypress/integration/models/providerVmware.ts:167]: Thanks a lot @sshveta for the Review !! 
The test case test_provider_update requirement was to edit all the three fields i.e Hostname, username and the password.
So, we have designed the test case accordingly.
And yes,  code can be modified if we just want to update any of the required fields. It's just a matter of using if-else or any other way out. But, as it wasn't a mandatory condition so, we implemented this way.

File: pkg/qe-tests/cypress/integration/models/providerVmware.ts [hunk 4]
```diff
@@ -153,16 +153,18 @@ export class ProviderVmware extends Provider {
...
     this.fillUsername(username);
     this.fillPassword(password);
     this.verifyCertificate();
     clickByText(addButtonModal, saveButton);
-    this.populate(providerData);
+    // this.populate(providerData);
```
[the remark this answers is not in the document]
Comment (kpunwatk) [reply pkg/qe-tests/cypress/integration/models/providerVmware.ts]: Thanks for noticing this, it is a required step by mistake I commented it :)

File: pkg/qe-tests/cypress/integration/tests/Rhv/editProvider.ts (hunk 2 above)
Comment (kpunwatk) [reply pkg/qe-tests/cypress/integration/tests/Rhv/editProvider.ts]: Modified the descibe :)

File: pkg/qe-tests/cypress/integration/models/providerVmware.ts [hunk 5]
```diff
@@ -153,9 +153,16 @@ export class ProviderVmware extends Provider {
     this.populate(providerData);
   }
 
-  edit(providerData: VmwareProviderData): void {
+  edit(providerData: VmwareProviderData, incorrectProviderData: VmwareProviderData): void {
```
Comment (ibragins) [inline pkg/qe-tests/cypress/integration/models/providerVmware.ts:156]: You shouldn't have additional provider data sent here, each time all fields are updated with the new name. Please remain sending only one provider data, if you want to put incorrect inputs on purpose - just do that.

**Review by ibragins — COMMENTED** [review]
Please update edit method according to my comments above

File: pkg/qe-tests/cypress/integration/models/providerVmware.ts (hunk 5 above)
Comment (kpunwatk) [reply pkg/qe-tests/cypress/integration/models/providerVmware.ts]: Thanks @ibragins, Please review !

**Review by ibragins — APPROVED** [review]
LGTM

**Pull request merged**